src/main/java/net/library/service/BookService.java

```diff
@@ -86,21 +87,44 @@ public BookItemIdDto borrowActionForAnyBookItem(final UUID bookId, final UUID us
      * If no available book item is found
      * a 'NOT FOUND' error is thrown.
      */
-//    @Lock(LockModeType.PESSIMISTIC_WRITE)
-    @Lock(LockModeType.OPTIMISTIC)
     @Transactional(isolation = Isolation.READ_UNCOMMITTED)
     public void borrowActionBookItemById(final UUID bookItemId, final UUID userId, final BookItemStatus bookItemStatus) {
         final var currentTime = Utils.currentDate();
```

Comment:
There is no official Java convention requiring final on local variables.
please remove final everywhere, and let’s keep our codebase consistent by avoiding it going forward